asgi_rabbitmq/core.py

```diff
@@ -65,34 +83,40 @@ def __init__(self, expiry, group_expiry, get_capacity, crypter, resolve):
             GROUP_DISCARD: self.group_discard,
             SEND_GROUP: self.send_group,
             DECLARE_DEAD_LETTERS: self.declare_dead_letters,
-            EXPIRE_GROUP_MEMBER: self.expire_group_member,
-            RECEIVE_TWISTED: self.receive_twisted,
         }
+        self.waiting_calls = []
+        self.messages = {}
+        self.message_number = 0
         self.are_confirmations_enabled = False
 
-    # Utilities.
-
-    def register_channel(self, method, amqp_channel):
-        """
-        Assign open AMQP channel to this protocol instance.  Call passed
-        method afterwards.  Used as AMQP channel open callback mostly.
-        """
+        self.amqp_channel = create_channel(self.apply_waiting)
```

Comment:
My first guess was that `create_channel` is a handler function here, but I see `Protocol` [is created](https://github.com/scrapinghub/asgi_rabbitmq/blob/98a0a948b29df268a8445671759507a878724be6/asgi_rabbitmq/core.py#L773-L775) with a channel object as a first arg, could you please clarifiy it?

Reply:
renamed the argument to channel_factory, and used it as a keyword argument when calling the constructor to make it a bit more clear that the value passed in is a factory method (despite the unfortunate name of the method in pika)